hr_cae_contract/models/hr_contract.py

```diff
@@ -321,30 +385,47 @@ def check_type_count(self):
 
     def create_amendment(self, type_id):
         self.ensure_one()
+        return self._create_contract("amendment", type_id)
+
+    def create_termination(self):
+        self.ensure_one()
+        type_id = self.env.ref("hr_cae_contract.hr_contract_type_termination")
+        return self._create_contract("termination", type_id)
+
+    def _create_contract(self, echelon, type_id):
+        self.ensure_one()
+        if self.has_termination_contract:
+            raise ValidationError(
+                _("This contract group already has a termination contract ")
+            )
         if not type_id:
             type_id = self.env["hr.contract.type"].search(
-                [("echelon", "=", "amendment")], limit=1
+                [("echelon", "=", echelon)], limit=1
             )
 
-        _logger.info("Creating %s Amendment" % type_id.name)
+        _logger.info(
+            "Creating Contract of echelon %s and type %s."
+            % (echelon, type_id.name)
+        )
 
         latest_contract_id = self.latest_contract_id
-        amendment = latest_contract_id.copy(
+        contract = latest_contract_id.copy(
             {
                 "state": "draft",
+                "echelon": echelon,
                 "type_id": type_id.id,
                 "duration": False,
                 "date_end": False,
                 "amendment_index": latest_contract_id.amendment_index + 1,
             }
         )
-        amendment.check_type_count()
+        contract.check_type_count()
```

Comment:
Why do you need to  launch manually this constraint?

Reply:
Because even though `type_id` is set on this copy and is also mentioned in the @api.constrains, it wasn't launched manually. I was puzzeled as well. Maybe I'm missing something? Or it's cleaned to overwrite the `copy`? Any Idea? Google doesn't seem to know.
The [documentation](https://www.odoo.com/documentation/12.0/reference/orm.html) (> `contrains`) doesn't mentions notes about `write` and `create` but doesn't mention anything about `copy`...

Reply:
Mmh, it's probably good to trigger it here then, good catch.